boxsdk/client/client.py

```diff
@@ -615,7 +611,7 @@ def create_retention_policy(
         """
         url = self.get_url('retention_policies')
         retention_attributes = {'policy_name': policy_name}
-        if retention_length == -1 or retention_length is None:
+        if retention_length == float('inf') or retention_length is None:
```

Comment:
I think if we're going to go this route, we should probably make `retention_length` a required parameter (and accept any `int` or `float('inf')`)